src/validation.cpp

```diff
@@ -3832,6 +3832,14 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
     if (ppindex)
         *ppindex = pindex;
 
+    if (!ActiveChainstate().IsInitialBlockDownload()) {
+        // These messages are valuable for detecting potential
+        // selfish mining behavior; if multiple displacing headers are seen
+        // near simultaneously across many nodes in the network, this might be
+        // an indication of selfish mining.
+        LogPrintf("saw new valid header hash=%s\n", hash.ToString());
```

Comment:
Maybe use `LogPrint(BCLog::VALIDATION,` when in IBD?

Reply:
Good call - done.